src/sentry/tagstore/v2/backend.py

```diff
@@ -735,8 +725,16 @@ def update_group_for_events(self, project_id, event_ids, destination_id):
             event_id__in=event_ids,
         ).update(group_id=destination_id)
 
-    def _get_environment_filter(self, environment_id):
+    def _get_environment_filter(self, environment_id, join=False):
```

Comment:
We talked about this out of band but noting it here too: it might make sense to change this method to something like: `_add_environment_filter(queryset, enviroment_id)` and have it use the `queryset.model` to determine the filter condition rather than having it provided here via `join`.